contracts/L2Deposit.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+import "./interfaces/IArbitrumL1GatewayRouter.sol";
+import "./interfaces/IArbitrumL1Inbox.sol";
+import "./interfaces/IL2Deposit.sol";
+import "./interfaces/IOptimismL1StandardBridge.sol";
+import "./interfaces/IPermanentStorage.sol";
+import "./interfaces/ISpender.sol";
+import "./utils/BaseLibEIP712.sol";
+import "./utils/Ownable.sol";
+import "./utils/L2DepositLibEIP712.sol";
+import "./utils/SignatureValidator.sol";
+
+contract L2Deposit is IL2Deposit, ReentrancyGuard, Ownable, BaseLibEIP712, SignatureValidator {
+    using SafeERC20 for IERC20;
+
+    // Peripherals
+    address public immutable userProxy;
+    IPermanentStorage public immutable permStorage;
+
+    // Bridges
+    IArbitrumL1GatewayRouter public immutable arbitrumL1GatewayRouter;
+    IArbitrumL1Inbox public immutable arbitrumL1Inbox;
```

Comment:
After removing collect refund function, I find that we could also remove the inbox reference 😎

Reply:
Removed in 560a1df